src/test/java/de/fraunhofer/aisec/cpg/frontends/LanguageFrontendTest.java

```diff
@@ -31,12 +31,25 @@
 import de.fraunhofer.aisec.cpg.TranslationConfiguration;
 import de.fraunhofer.aisec.cpg.TranslationManager;
 import de.fraunhofer.aisec.cpg.TranslationResult;
+import de.fraunhofer.aisec.cpg.graph.TypeManager;
+import de.fraunhofer.aisec.cpg.graph.type.TypeParser;
 import java.io.File;
 import java.util.concurrent.ExecutionException;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 class LanguageFrontendTest {
 
+  /**
+   * {@link TypeParser} and {@link TypeManager} hold static state. This needs to be cleared before
+   * all tests in order to avoid strange errors
+   */
+  @BeforeEach
+  void resetPersistentState() {
```

Comment:
Minor suggestion: Instead of having the resetPersistentState function copy/pasted into all the tests that need it can we introduce an abstract test class and implement the function there? @BeforeEach should then propagate to the derived class as well. This will make it easier if something like this needs to be done in the future for other subsystems as well.

Reply:
Ah does that work? Didn't know whether this propagates

Reply:
I am 99% sure it does :)

Reply:
Just tested it, it does :+1: